### ISSUE TITLE
Using this package when the dataset is big, problem...

### ISSUE THREAD

**Issue by NoeCors** [andrewraim/COMMultReg#3]
Sorry to bother you,
It may happen that the function "out_cmm = cmm_reg_raw(y = y, X = X, W = W, control = ctrl)" do not work when the dataset is "big"? Like 50 columns and 100 rows. Because I have tried the code with the dataset cars provided by R and it works but when I use my dataset (I leave everything else untouched) I get that R crashes....
It doesn't give me any errors.. it just crashes...

Thanks in advance

**Comment by andrewraim** [comment]
Dear @NoeCors,

There are some known limitations on the sizes of the problems this package can currently handle. If, say, one of your observations has `k` categories and `m` overall trials, the size of the multinomial sample space is `choose(m+k-1, m)`. We currently have to iterate through that sample space several times to fit the model. This is a limitation of the initial work, but something we hope to improve to fit data with more categories and larger numbers of trials.

That sounds like it could definitely be a problem if you are using 50 columns. If you could post a minimal working example where R crashes for you, I could verify that it is the issue and see if there is a way to handle it more gracefully.

**Comment by andrewraim** [comment]
Thanks for sending over the example. Is your `x.zero` variable supposed to be `x`? If so, it looks like the sample space issue is indeed the problem. For example, the first row in `x` has `m = 250` trials and `k = 50` categories, so the size of its sample space is
```r
R> m = rowSums(x)
R> k = ncol(x)
R> choose(m[1] + k - 1, m[1])
[1] 5.187887e+56
```
As you can see, that's a pretty large number. Iterating through that many points is intractable, so our current approach will not work with your dataset unfortunately. We would need a method that does not need to iterate over the sample space.

I see that the call to `cmm_reg_raw` will not allow itself to be interrupted, so the whole process needs to be killed to stop it. I will consider adding a periodic check for user interrupts in the C++ code, and some kind of warning if the number of iterations will be extremely large. I may not be able to get to it right away, but will take care of it next time I'm in the code.

**Comment by NoeCors** [comment]
Thank you very much for the reply, of course now I see the problem!
I will try to use a littler dataset.

**Comment by NoeCors** [comment]
Just another information, as far as you know (since you have already used it) the same problem is present when considering the multiplicative multinomial right?

**Comment by andrewraim** [comment]
Yes, that's correct - the estimation method we used in the CMM paper ("the Poisson regression trick") was inspired by the MM package <https://CRAN.R-project.org/package=MM> for the multiplicative multinomial model. I believe they also need to iterate over the whole sample space for the trick to work.
